File: beginner/__3000__/3301/3301.php
<?php 

    list($a, $b, $c) = explode(" ", readline());

    $a = (int) $a;
    $b = (int) $b;
    $c = (int) $c;

    $data = [$a, $b, $c];
    $sobrinhos = ["huguinho", "zezinho", "luisinho"];

    function get_middle_index($list){
        $mmin = min($list);
        $mmax = max($list);

        for($i = 0; $i < 3; $i++){
            if($list[$i] != $mmin && $list[$i] != $mmax){
                return $i;
            }
        }
        return -1;
    }

    $response = $sobrinhos[get_middle_index($data)];

    echo "$response\n";

?>

File: beginner/__3000__/3301/3301.py
entrada = [int(x) for x in input().split()]

def get_middle_element_index(list):

    mmax = max(list)
    mmin = min(list)

    for i in range(3):
        if list[i] != mmin and list[i] != mmax:
             return i
    return -1


sobrinhos = ["huguinho", "zezinho", "luisinho"]

print(sobrinhos[get_middle_element_index(input)])